test/integration/dnsrecord/dnsrecord_test.go

```diff
@@ -0,0 +1,294 @@
+		}
+		cluster = &extensionsv1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testName,
+			},
+		}
+	)
+
+	BeforeSuite(func() {
+		repoRoot := filepath.Join("..", "..", "..")
+
+		// enable manager logs
+		logf.SetLogger(zap.New(zap.UseDevMode(true), zap.WriteTo(GinkgoWriter)))
+
+		log := logrus.New()
+		log.SetOutput(GinkgoWriter)
+		logger = logrus.NewEntry(log)
+
+		By("starting test environment")
+		testEnv = &envtest.Environment{
+			UseExistingCluster: pointer.BoolPtr(true),
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				Paths: []string{
+					filepath.Join(repoRoot, "example", "20-crd-dnsrecord.yaml"),
+					filepath.Join(repoRoot, "example", "20-crd-cluster.yaml"),
+				},
+			},
+		}
+
+		cfg, err := testEnv.Start()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cfg).ToNot(BeNil())
+
+		By("setting up manager")
+		mgr, err := manager.New(cfg, manager.Options{
+			MetricsBindAddress: "0",
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(extensionsv1alpha1.AddToScheme(mgr.GetScheme())).To(Succeed())
+		Expect(awsinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
+
+		Expect(dnsrecordctrl.AddToManager(mgr)).To(Succeed())
+
+		var mgrContext context.Context
+		mgrContext, mgrCancel = context.WithCancel(ctx)
+
+		By("starting manager")
+		go func() {
+			err := mgr.Start(mgrContext)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		// test client should be uncached and independent from the tested manager
+		c, err = client.New(cfg, client.Options{
+			Scheme: mgr.GetScheme(),
+			Mapper: mgr.GetRESTMapper(),
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(c).NotTo(BeNil())
+
+		flag.Parse()
+		validateFlags()
+
+		awsClient, err = awsclient.NewClient(*accessKeyID, *secretAccessKey, aws.DefaultDNSRegion)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterSuite(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+
+		By("running cleanup actions")
+		framework.RunCleanupActions()
+
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	It("should successfully create and delete", func() {
+		By("setting up shoot environment")
+		Expect(c.Create(ctx, namespace)).To(Succeed())
+		Expect(c.Create(ctx, secret)).To(Succeed())
+		Expect(c.Create(ctx, cluster)).To(Succeed())
+		framework.AddCleanupAction(func() {
+			By("tearing down shoot environment")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+			Expect(client.IgnoreNotFound(c.Delete(ctx, secret))).To(Succeed())
+			Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+		})
+
+		By("creating dnsrecord")
+		dns := newDNSRecord(testName)
+		Expect(c.Create(ctx, dns)).To(Succeed())
+		framework.AddCleanupAction(func() {
+			By("deleting dnsrecord")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, dns))).To(Succeed())
+
+			By("waiting until dnsrecord is deleted")
+			Expect(extensions.WaitUntilExtensionObjectDeleted(
+				ctx,
+				c,
+				logger,
+				dns.DeepCopy(),
```

Comment:
```suggestion
				dns,
```

Reply:
Without `DeepCopy()` the test fails. The method resets the resource and the subsequent `verify` doesn't have the right values, so it fails. Could perhaps be avoided in other ways, but the `DeepCopy` here is perhaps the most convenient one.

Reply:
Ah, got it :)